tools/wdio/plugins/browser-matcher.mjs

```diff
@@ -0,0 +1,43 @@
+/**
+ * This is a WDIO worker plugin that provides a global method allowing for the
+ * filtering of tests by a browser match.
+ */
+export default class BrowserMatcher {
+  #browserSpec
+
+  async beforeSession (_, capabilities) {
+    this.#browserSpec =
+      `${this.#getBrowserName(capabilities)}@${capabilities.browserVersion || capabilities.version}`
+    global.test = this.#test.bind(this)
+  }
+
+  #test (matcher) {
```

Comment:
I like the higher order function approach. I think this would be more clear if instead of `test` we had `inBrowsers` or `withBrowsersMatching` or similar. Seems like `test` is not very descriptive. I don't feel strongly since it's documented well in your .md, but something to consider if you want.

Reply:
I like that idea. What does everyone else think? @metal-messiah @cwli24

Reply:
Agreed, `test` overlaps too much with the other mocha/jesty type command syntax